samwise.js

```diff
@@ -1,47 +1,42 @@
 /**
```

Comment:
SCRIPT-I like how you've used promise chaining here, it makes the code flow very clear.